New ICCG-ship & related content (missiles, sensors tweaks)

ImJustKisik opened the pull request:
# Описание

> **Разведывательный корабль Флота Дальнего Плавания ГКК добавляется вместе с новыми ракетами. Это современное, оснащённое передовой техникой гильгамешское судно, которое будет бороздить космос сектора где сейчас пролетает ИКН "Сьерра".
Данное судно может быть использована как база для исследовательских миссий, которые ведёт Гильгамеш на фронтире с целью добраться до тайны потери ГЭК "Магнус" их оппонентами, а также с целью открытия новых залежей форона и пригодных для колонизации планет. 
Но само собой, мирными инициативами дело может и не ограничиться, ведь у вас готовы к пуску ракеты из двух аппаратов, а в секторе можете быть заинтересованы не только вы, но и ЦПСС. Не говоря о таких игроках, как НТ. Опробуйте новые механики, не доверяйте никому и исследуйте космос во славу вашей Родины!**

## Скриншоты

![Z1Ship](https://user-images.githubusercontent.com/87862013/193846020-4a299147-479f-4913-90c9-ac611afdae69.png)
![Z2Ship](https://user-images.githubusercontent.com/87862013/193846029-271100d9-58f0-48dd-9e5c-83521147747d.png)




Content: 
---

> **"Пушка! Они заряжают пушку! Зачем? А! Они будут стрелять!"**

Добавляются ракеты которые могут взаимодействовать с овермапой, в том числе, могут перемещаться между различными Z-уровнями и овермапой.
Ракеты полностью модульные, можно устанавливать различные двигатели, боеголовки и другое оборудование по своему усмотрению. 

Ракеты состоят из: 

- Модуля
- Дополнительного оборудования (необязательно)
- Ускорителя


| Модули     |   |
| ------------- | -----:|
| Взрывной (explosive charge)    | Взрывает.|
| Антифазовый (dynamic antiphase emitter)    |  Отключает щиты. |
| Противоракетный (PREDATOR missile disabler system)  |    Подрывает другие ракеты. |
| Ядерный (nuclear warhead)  |    Взрывает, но очень сильно. (Щитспавн онли) |
| Пустотный (void warhead)  |    Взрывает, но немного слабее чем "очень сильно". (Щитспавн онли) |

---
| Ускорители     |   |
| ------------- | -----:|
| Стандартный (missile booster)    | Позволяет наводиться на корабли в радиусе действия сенсоров.|
| Противоракетный (HUNTER warp booster)    |   Позволяет наводиться на ракеты в радиусе действия сенсоров. |
| Наводящийся (pointman missile booster)  |    Наводится по координатам. (Установлен в ядерных и VOID-ракетах) |
| Планетарный (planetary missile booster)  |    Наводится по координатам на планету. |
---
| Прочее|   |
| ------------- | -----:|
| Автозаряд (automatic missile activator)    | Автоматически подрывает ракеты при касании цели.|
| Вместилище пассажиров (COMFMASTER Mk3 passenger compartment)   |   Позволяет полететь на ракете. Буквально. |


---
# Переработка сенсоров
Сенсоры больше не используют освещение (туман войны) для скрытия объектов на овермапе. 
Теперь для обнаружения другого корабля необходима идентификация, скорость идентификации зависит от: 
мощности сенсоров, скорости идентифицируемого корабля и расстояния между кораблями.
После обнаружения, корабль можно просканировать. Теперь сканы выглядят таким образом, отображая больше информации о корабле: 
![image](https://user-images.githubusercontent.com/87862013/193850322-0f0dab60-95c7-492d-9d30-8e1444735ec6.png)


Ивенты на овермапе, такие как ионные облака, требуют, чтобы они находились в зоне видимости сенсоров, но в остальном они также работают как и раньше.

---
# Новый сигнал бедствия 
Добавляется физическая кнопка сигнала бедствия, теперь она работает гораздо лучше! 
На все корабли на овермапе передаётся сообщение:
![image](https://user-images.githubusercontent.com/87862013/193852705-7d3c18c9-d8e1-4d8c-b98a-dd0667bb90d4.png)
Также на овермапе появляется специальный ивент, который видно даже с выключенными сенсорами! 

Rig and ICCG sprites by ЭллГуд 

Missiles code from Bay12/BoH-Bay

---

:cl:
rscadd: Adds new away ship (w.i.p) 
rscadd: Overmap capable missiles have been added:
rscadd: Armed missiles enter the overmap when they reach the edge of the map. They may or may not travel along the overmap based on what the installed equipment does. Upon entering a overmap tile, it may enter a random z level on the tile and travel through it (i.e. asteroid fields), or impact immediately (i.e. on planets)
rscadd: Missiles are usually fitted with automatic arming devices.
rscadd: Missiles detonate when they are armed and bump into a solid object (Note that you can still bump into the missile safely. It detonates upon bumping into something itself)
rscadd: Missiles are modular. By using a screwdriver on a missile you can open a maintenance panel to install equipment, or use a crowbar on an open missile to remove equipment. Depending on the specific equipment, you may be able to interact further with it
tweak: Sensors no longer uses lighting for obfuscation, but instead casts images of overmap objects to the client screen upon identification. Ships requiring scanning ships to remain within range until full identification is complete. 
/:cl:

ImJustKisik committed:
ракеты мои ракеты

overmap flags pt 1

Missiles, baby

admin missiles and misc stuff

Update thruster.dm

sonar like sensors

classes

changes da

skyrat connected computers

disstres button

Identify me, scanner-sempai~

я хочу умереть

update update

Feat: ship traits (#337)

ракетки мои ракетки

ImJustKisik committed:
Content

cigs

RPG for ICCG

конфликты иконок обожаю

ImJustKisik committed:
New guns

RPG for ICCG

конфликты иконок обожаю

cigs

Update fancy.dm

Brand new guns

Den109G commented on code/modules/overmap/projectiles/missiles/equipment/missile_equipment.dm:
```diff
@@ -0,0 +1,35 @@
+
+// Called by the missile when it detonates
+/obj/item/missile_equipment/proc/on_trigger(var/atom/triggerer)
+	return
```
![image](https://user-images.githubusercontent.com/87372121/192087707-94f9dcfc-0f4e-45d1-b45a-901d5475f65a.png)

ImJustKisik committed:
Revert "Content"

This reverts commit 106263e914fba5c41135c8e90060eb2ecd4a9a92.

ImJustKisik committed:
ICCGN faction

Pulls terran stuff out of maps/torch, gives it a facelift and sets it up in the packs/faction_iccgn dir to help with multi map inclusion.
Adds iccgn patches, badges, ranks, utility jacket

ImJustKisik committed:
Yeah, squash it~

Pizdec

ракеты мои ракеты

overmap flags pt 1

Missiles, baby

admin missiles and misc stuff

Update thruster.dm

sonar like sensors

classes

changes da

skyrat connected computers

disstres button

Identify me, scanner-sempai~

я хочу умереть

update update

Feat: ship traits (#337)

ракетки мои ракетки

Update sensors.dm

апдейты фиксы да

doors plus decor

ракеты мои ракеты

overmap flags pt 1

Missiles, baby

admin missiles and misc stuff

Update thruster.dm

sonar like sensors

classes

changes da

skyrat connected computers

disstres button

Identify me, scanner-sempai~

я хочу умереть

update update

Feat: ship traits (#337)

ракетки мои ракетки

Update sensors.dm

апдейты фиксы да

doors plus decor

cigs

RPG missile launcher

RPG for ICCG

конфликты иконок обожаю

Content

cigs

RPG for ICCG

конфликты иконок обожаю

Update missile_equipment.dm

Update overmap.dmi

Revert "Content"

This reverts commit 106263e914fba5c41135c8e90060eb2ecd4a9a92.

desc's

sensors updates

another_portion_of_fixes

desc's

sensors updates

Merge branch 'ship' into pizdec

Merge pull request #10 from LordNest/pizdec

Pizdec

SNZ_fix

Merge branch 'pizdec' of https://github.com/LordNest/Baystation12 into pizdec

Minor_fixes

Я_ЭТО_Починил

Merge pull request #11 from LordNest/pizdec

SNZ_teper'_rabotaet

sexual perversions with smes

Some emissive objects now have overlays that light over the lighting plane.

da

sexual perversions with smes

Some emissive objects now have overlays that light over the lighting plane.

da

fix

Merge pull request #6 from ImJustKisik/pizdec

Pizdec-2

SNZ_fixes

Merge pull request #12 from LordNest/pizdec

SNZ_Incident

TCOMS

Merge pull request #7 from ImJustKisik/ship

Ship

clothing_fixes

ICCGN faction

Pulls terran stuff out of maps/torch, gives it a facelift and sets it up in the packs/faction_iccgn dir to help with multi map inclusion.
Adds iccgn patches, badges, ranks, utility jacket

Crew_and_hangar_fixes

По причине того, что злоебучие шлюзы отказываются воспринимать шаттл, шаттл был убран в ангар, а над мостиком надстроена подстилка под ангар

Merge pull request #13 from LordNest/pizdec

SNZ_Hangar

Update vending.dm

ImJustKisik committed:
Yeah, squash it~

Pizdec

ракеты мои ракеты

overmap flags pt 1

Missiles, baby

admin missiles and misc stuff

Update thruster.dm

sonar like sensors

classes

changes da

skyrat connected computers

disstres button

Identify me, scanner-sempai~

я хочу умереть

update update

Feat: ship traits (#337)

ракетки мои ракетки

Update sensors.dm

апдейты фиксы да

doors plus decor

ракеты мои ракеты

overmap flags pt 1

Missiles, baby

admin missiles and misc stuff

Update thruster.dm

sonar like sensors

classes

changes da

skyrat connected computers

disstres button

Identify me, scanner-sempai~

я хочу умереть

update update

Feat: ship traits (#337)

ракетки мои ракетки

Update sensors.dm

апдейты фиксы да

doors plus decor

cigs

RPG missile launcher

RPG for ICCG

конфликты иконок обожаю

Content

cigs

RPG for ICCG

конфликты иконок обожаю

Update missile_equipment.dm

Update overmap.dmi

Revert "Content"

This reverts commit 106263e914fba5c41135c8e90060eb2ecd4a9a92.

desc's

sensors updates

another_portion_of_fixes

desc's

sensors updates

Merge branch 'ship' into pizdec

Merge pull request #10 from LordNest/pizdec

Pizdec

SNZ_fix

Merge branch 'pizdec' of https://github.com/LordNest/Baystation12 into pizdec

Minor_fixes

Я_ЭТО_Починил

Merge pull request #11 from LordNest/pizdec

SNZ_teper'_rabotaet

sexual perversions with smes

Some emissive objects now have overlays that light over the lighting plane.

da

sexual perversions with smes

Some emissive objects now have overlays that light over the lighting plane.

da

fix

Merge pull request #6 from ImJustKisik/pizdec

Pizdec-2

SNZ_fixes

Merge pull request #12 from LordNest/pizdec

SNZ_Incident

TCOMS

Merge pull request #7 from ImJustKisik/ship

Ship

clothing_fixes

ICCGN faction

Pulls terran stuff out of maps/torch, gives it a facelift and sets it up in the packs/faction_iccgn dir to help with multi map inclusion.
Adds iccgn patches, badges, ranks, utility jacket

Crew_and_hangar_fixes

По причине того, что злоебучие шлюзы отказываются воспринимать шаттл, шаттл был убран в ангар, а над мостиком надстроена подстилка под ангар

Merge pull request #13 from LordNest/pizdec

SNZ_Hangar

Update vending.dm

ImJustKisik committed:
Overmap updates

Update sierra_overmap.dm

Update sierra_overmap.dm

Update sierra_overmap.dm

ImJustKisik committed:
icons fix

ImJustKisik committed:
ofd droppod

ImJustKisik committed:
t-coms fix

LordNest committed:
Rig plus away site's fix

RIG sprites fixed and vessel mass increased in order to lower it's speed

RIG_Fixes

Rig temporary icons+textes with lore of expedition

Gagarin_class

OFD_console_removal

ImJustKisik committed:
Merge remote-tracking branch 'upstream/master220' into pizdec

ImJustKisik committed:
Admin stuff

ImJustKisik committed:
Overmap radio

ImJustKisik committed:
sensors update

ImJustKisik committed:
update

ImJustKisik committed:
Merge branch 'pizdec' of https://github.com/ImJustKisik/Baystation12 into pizdec

ImJustKisik committed:
cigs desc update

ImJustKisik committed:
Map fix plus new terran drinks

ImJustKisik committed:
Small map fixes plus drinks

Map fix plus new terran drinks

map update

ImJustKisik committed:
Update farfleet-1.dmm

ImJustKisik committed:
Update vending.dm

ImJustKisik committed:
Merge branch 'pizdec' of https://github.com/ImJustKisik/Baystation12 into pizdec

ImJustKisik committed:
Merge remote-tracking branch 'upstream/master220' into pizdec

ImJustKisik committed:
map update

ImJustKisik committed:
doors opacity fix

ImJustKisik committed:
Name and Icon update

ImJustKisik committed:
Merge remote-tracking branch 'upstream/master220' into pizdec

ImJustKisik committed:
Fixes

ImJustKisik committed:
Speed and accees fixes

ImJustKisik committed:
Merge remote-tracking branch 'upstream/master220' into pizdec

ImJustKisik committed:
Был бы ты Южин человек

ImJustKisik committed:
Update baystation12.dme

ImJustKisik committed:
Combat turn and combat roll

ImJustKisik committed:
fixes

ImJustKisik committed:
Update farfleet.dm

ImJustKisik committed:
fixes

ImJustKisik committed:
fixes from testmerge

ImJustKisik committed:
Update farfleet-1.dmm

ImJustKisik committed:
map update

ImJustKisik committed:
RPG refactor

LordNest committed:
Mapfixes

ImJustKisik committed:
Merge pull request #21 from LordNest/Mapxix

Mapfixes

ImJustKisik committed:
sensor range now bound w/ consoles

ImJustKisik committed:
lordnest map fixes

ImJustKisik committed:
Merge remote-tracking branch 'upstream/master220' into pizdec
